### ISSUE TITLE
Apache cannot be installed because of PPA for PHP and Apache

### ISSUE THREAD

**Issue by psihius** [fideloper/Vaprobash#439]
```bash
==> Vaprobash: Reading package lists...
==> Vaprobash: E
==> Vaprobash: : 
==> Vaprobash: Unable to correct problems, you have held broken packages.
==> Vaprobash: >>> Configuring Apache
==> Vaprobash: sudo: a2dismod: command not found
==> Vaprobash: sudo: a2dismod: command not found
==> Vaprobash: sudo: a2enmod: command not found
==> Vaprobash: /usr/local/bin/vhost: line 177: /etc/apache2/sites-available/192.168.22.10.xip.io.conf: No such file or directory
==> Vaprobash: /usr/local/bin/vhost: line 181: /etc/apache2/sites-available/192.168.22.10.xip.io.conf: No such file or directory
==> Vaprobash: /usr/local/bin/vhost: line 185: cd: /etc/apache2/sites-available/: No such file or directory
==> Vaprobash: apache2: unrecognized service
==> Vaprobash: sudo: a2dissite: command not found
==> Vaprobash: sudo: a2enmod: command not found
==> Vaprobash: apache2: unrecognized service
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Diving in into VM and trying the command from scripts/apache.sh

```bash
vagrant@vaprobash:~$ sudo apt-get install apache2 apache2-mpm-event
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 apache2 : PreDepends: dpkg (>= 1.17.14) but 1.17.5ubuntu5.3 is to be installed
           Recommends: ssl-cert but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
vagrant@vaprobash:~$ 
```

The dependencies for the package are off, and as far as I was able to try and solve it - nothing came of it.

For now I just forked the repo and made modifications, also adding XHProf and cachefilesd for NFS :) (I think i'm gonna make some pull requests pretty soon :) )

**Comment by fideloper** [comment]
Thanks - can you show me your Vagrantfile? Are you using vmware provider?

What modifications did you make to fix?

Thanks!

**Comment by psihius** [comment]
@fideloper 
It's the default Vagrantfile from the repo, the only thing I did is uncommented Apache provisioning.

I'm on Virtualbox
Well, I didn't fix it - I just removed the PPA repository from the PHP and Apache scripts and installing what comes with the Ubuntu 14.04 (apache 2.4.7 and PHP 5.5.9 - PPA repository has Apache 2.4.12 and PHP 5.6.x)

**Comment by psihius** [comment]
At the same time, the template for the Apache host does not seem to work with PHP-FPM - the 
```apache
<FilesMatch \.php$>
            # Change this "proxy:unix:/path/to/fpm.socket"
            # if using a Unix socket
            SetHandler "proxy:fcgi://127.0.0.1:9000"
        </FilesMatch>
```
does not match any .php files I tried to open and I get PHP code in the browser, although I just switched to nginx and didn't bother much :)

**Comment by fideloper** [comment]
Thanks, I'll check this out and see what I find.

**Comment by xus** [comment]
Hi guys, I have the same issue.  I just uncommented the apache provisioning in the default vagrantfile.
Did you find something to fix this issue?

**Comment by cambri** [comment]
Issue seems to be with apache2-mpm-event.  Causing held package dependency issues on Ubuntu 14.04.  I commented out and provisioning proceeds, now what are the effects of removing...

**Comment by fideloper** [comment]
mpm-event I think will be the default in any case since we never install libapache2-mod-php5 so I'll bet that's just fine.

**Comment by cambri** [comment]
1. Selected php 5.5 in vapro config, as I've had similar issues with 5.6 on another server
2. Commented out apache2-mpm-event
3. Added "sudo a2enmod proxy_fcgi" to apache.sh per these directives https://serversforhackers.com/proxying-to-fastcgi-in-ubuntuapache-24
4. 403 errors until I added index.html/index.php, in webroot
5. Confirmed with phpinfo() that apache is using fpm = (Server API FPM/FastCGI)
This should help others that are having issues.  I made some assumptions about step 3, as the apache.sh and vhost file already have most of the the proxy implementation in place.  It seemed reasonable that this might be needed.

**Comment by fideloper** [comment]
Sounds legit! I'll hopefully get a fix soon, altho that sounds like enough
info for someone to help with a PR fairly easily :D

**Comment by chrispelzer** [comment]
There was definitely a change that happened with apache2-mpm-event.

I ran into this the other day with trying to rebuild our development box on a forked version of Vaprobash12, which caused me to finally look into updating our box.

Haven't noticed any issues with removing it so far, but not done yet. Once I'm finished I'll put in a PR, unless someone beats me to it.

**Comment by psihius** [comment]
Yes, same here. PPA  version of apache just does not go together with the apache2-mpm-event package. I just rebuilt my box with nginx instead though.
Hope the next ubuntu version bumps the apache2 version enought that we dont need the PPA

**Comment by xus** [comment]
Perfect, now works fine thanks to cambri changes. Thanks a lot guys.

**Comment by jahvi** [comment]
I'm still getting this issue even when I deleted the ondrej apache2 ppa, have anyone else found a workaround?

**Comment by damianlewis** [comment]
Same problem here. Switched to Nginx until the issue with Apache is fixed.

**Comment by fideloper** [comment]
Should be resolved via #489
